mindsdb_native/libs/phases/data_extractor/data_extractor.py

```diff
@@ -88,6 +91,43 @@ def _get_prepared_input_df(self):
                 # if no data frame yet, make one
                 df = self._data_from_when()
 
+            if self.transaction.lmd['setup_args'] is not None and self.transaction.lmd['tss']['is_timeseries']:
+                if 'make_predictions' not in df.columns:
+                    df['make_predictions'] = [True] * len(df)
+
+                setup_args = deepcopy(self.transaction.lmd['setup_args'])
+
+
+                if len(df) > 1 and self.transaction.lmd['tss']['group_by'] is not None:
+                    encountered_set = set()
+                    unique_group_by_rows = []
+                    for i in range(en(df)):
+                        val_str = ''
+                        for group_col in self.transaction.lmd['tss']['group_by']:
+                            val_str += '###-------------###' + str(df.iloc[i][group_col])
+                            if val_str not in encountered_set:
+                                encountered_set.add(val_str)
+                                unique_group_by_rows.append(df.iloc[i])
+                else:
+                    unique_group_by_rows = [df.iloc[0]]
+
+                historical_df = None
+
+                for row in unique_group_by_rows:
+                    setup_args['query'] = create_history_query(setup_args['query'], self.transaction.lmd['tss'], self.transaction.lmd['stats_v2'], row)
+
+                    if historical_df is None:
```

Comment:
this is always `True`

Reply:
Hmh, no, `historical_df` gets assigned a value in the first itteration of the loop and then the other dataframes are concated to it afterwards.